grails-app/controllers/org/pih/warehouse/order/OrderController.groovy

```diff
@@ -750,19 +750,21 @@ class OrderController {
             orderItem.productSupplier = productSupplier
         }
 
+        if (!order.save(flush:true)) {
+            throw new ValidationException("Order is invalid", order.errors)
```

Comment:
@jmiranda I moved this out of the try-catch, to get the advantage of ErrorController's `handleValidationErrors` (which was missed within the `catch`)